src/snowflake/snowpark/functions.py

```diff
@@ -8700,3 +8701,123 @@ def make_interval(
             nanoseconds,
         )
     )
+
+
+def map(
+    dataframe,
+    func: Callable,
+    output_types: List[StructType],
+    output_column_names: Optional[List[str]] = None,
+    wrap_row=True,
+    **kwargs: Dict[str, Any],
+):
+    """Returns a new DataFrame with the result of applying `func` to each of the
+    rows of the specified DataFrame.
+
+    Args:
+        dataframe: The DataFrame instance.
+        func: A function to be applied to every row of the DataFrame.
+        output_types: A list of types for values generated by the ``func``
+        output_column_names: A list of names to be assigned to the resulting columns.
+        wrap_row: A flag to determine if the current row must be wrapped in an instance of the :class:`Row` class.
+        kwargs: Additional arguments to register the UDTF used to apply the Python function.
+
+    Example 1::
+
+        >>> from snowflake.snowpark.types import IntegerType
+        >>> from snowflake.snowpark.functions import map
+        >>> df = session.create_dataframe([[10, "a"]], schema=["col1", "col2"])
+        >>> new_df = map(df, lambda row: row[0] * row[0], output_types=[IntegerType()])
+        >>> new_df.show()
+        ---------
+        |"C_1"  |
+        ---------
+        |100    |
+        ---------
+        <BLANKLINE>
+
+    Example 2::
+
+        >>> new_df = map(df, lambda row: (row[1], row[0] * 3), output_types=[StringType(), IntegerType()])
+        >>> new_df.show()
+        -----------------
+        |"C_1"  |"C_2"  |
+        -----------------
+        |a      |30     |
+        -----------------
+        <BLANKLINE>
+
+    Example 3::
+
+        >>> new_df = map(df, lambda row: (row[1], row[0] * 3), output_types=[StringType(), IntegerType()], output_column_names=['col1', 'col2'])
+        >>> new_df.show()
+        -------------------
+        |"COL1"  |"COL2"  |
+        -------------------
+        |a       |30      |
+        -------------------
+        <BLANKLINE>
+
+    Example 4::
+        >>> df = session.create_dataframe([10, 20, 30])
+        >>> new_df = map(df, lambda x: x * x, output_types=[IntegerType()], wrap_row=False)
+        >>> new_df.sort(new_df[0]).show()
+        ---------
+        |"C_1"  |
+        ---------
+        |100    |
+        |400    |
+        |900    |
+        ---------
+        <BLANKLINE>
+
+    Note:
+        The result of the `func` function must be either a scalar value or a tuple containing the same number of elements
+        as specified in the `output_types` argument.
+    """
+    if "packages" not in kwargs:
+        kwargs["packages"] = ["snowflake-snowpark-python"]
```

Comment:
We need to include the `snowflake-snowpark-python` package because the handler uses the `snowflake.snowpark.Row` class . See the `_MapFunc` handler .

If we don't include this package we get this error: `ModuleNotFoundError: No module named 'snowflake'`

Reply:
Ok, this change is applied on the last commit.